dv/uvm/icache/data/ibex_icache_testplan.hjson

```diff
@@ -7,24 +7,95 @@
   entries: [
```

Comment:
How is error handling going to be tested (both PMP errors and "IMEM" errors)? Will there be explicit tests for the error lines, or will those be toggled during all of these tests?

In addition, how will resets be tested?

Reply:
In all tests, some memory regions will return errors (PMP or actual bus errors). The scoreboard will be checking they are handled correctly. So I think this comes "for free".

About resets: Good point, I haven't thought about that. Do we have any documentation about what we're actually testing for with them? (After all, the world comes up with all signals as 'X', so in some sense we do a reset test every time!)

Reply:
Re error testing: Got it. Just a suggestion, what do you think about disabling errors in the sanity test (to make it as "sane" as possible) and creating a new error stressing test where all errors are asserted much more frequently? I'm assuming that in the rest of the tests errors won't be asserted super frequently (but correct me if I'm wrong).
Maybe another interesting scenario would be having an error asserted during invalidation.

Re reset testing: It sort of varies per IP, for example in Ibex when I test resets I just make sure the core can reset, jump to the start address, and finish the test without much other checking. For the Icache there would be some more security-related implications (we don't want old data being accessible after a reset clears/invalidates everything), so just going off of the existing documentation my guess is just that you would want to reset the cache and then guarantee that everything gets invalidated afterwards.